flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -119,7 +119,7 @@ public void testLimitPushDown() {
         Row.of(2, "b", 20.0),
         Row.of(3, null, 30.0)
     );
-    Assert.assertEquals("Should produce the expected records", expectedList, resultExceed);
+    assertSameElements(expectedList, resultExceed);
```

Comment:
How about keeping this description by `#describeAs(...)`

Reply:
For AssertJ, the assertion message is already pretty good. I updated one test so it would fail and here's the output I got. I think it's pretty clean / self explanatory.

```
Expecting:
  [+I[1, iceberg, 10.0], +I[2, b, 20.0]]
to contain exactly in any order:
  [+I[1, iceberg, 10.0]]
but the following elements were unexpected:
  [+I[2, b, 20.0]]
```

With `.as` before the `containsExactlyInAnyOrderElementsOf`, here's the output. I don't personally think it provides much value.

```
java.lang.AssertionError: [should contain the same elements] 
Expecting:
  [+I[1, iceberg, 10.0], +I[2, b, 20.0]]
to contain exactly in any order:
  [+I[1, iceberg, 10.0]]
but the following elements were unexpected:
  [+I[2, b, 20.0]]
```

Reply:
I think we should make the assertSameElements open(can be optional) for description.
For the case you mentioned, i think it didn't provide much value because the statement itself, not for all other descriptions.
WDYT @kbendick ?

Reply:
So you mean add a description argument like `assertSameElements(String errorMessage, List<Row> exepected, List<Row> actual)`?

I'm happy to add that as well as keep the original signature. I can't think of a place in the current file that it would be useful as they're all the same scenarios, but I'm happy to add it.

Reply:
Added that signature in should anybody want to use it @yittg.